hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadataWriter.java

```diff
@@ -837,10 +840,75 @@ public void update(HoodieCleanMetadata cleanMetadata, String instantTime) {
+    dirInfoList.clear();
+
+    LOG.info("Restoring MDT to " + restoreToInstantTime + " at " + instantTime);
+    getWriteClient().restoreToInstant(restoreToInstantTime, false);
+
+    // At this point we have also reverted the cleans which have occurred after the restoreToInstantTime. Hence, a sync
+    // is required to bring back those cleans.
+    try {
+      initTableMetadata();
+      HoodieCleanMetadata cleanMetadata = new HoodieCleanMetadata();
+      Map<String, HoodieCleanPartitionMetadata> partitionMetadata = new HashMap<>();
+      for (String partition : metadata.fetchAllPartitionPaths()) {
+        FileStatus[] metadataFiles = metadata.getAllFilesInPartition(new Path(dataWriteConfig.getBasePath(), partition));
+        if (!dirInfoMap.containsKey(partition)) {
+          // Entire partition has been deleted
+          List<String> filePaths = Arrays.stream(metadataFiles).map(f -> f.getPath().getName()).collect(Collectors.toList());
+          HoodieCleanPartitionMetadata cleanPartitionMetadata = new HoodieCleanPartitionMetadata(partition, "", filePaths, filePaths,
+              Collections.emptyList(), true);
+          partitionMetadata.put(partition, cleanPartitionMetadata);
+        } else {
+          // Some files cleaned in the partition
+          Map<String, Long> fsFiles = dirInfoMap.get(partition).getFileNameToSizeMap();
+          List<String> filesDeleted = Arrays.stream(metadataFiles).map(f -> f.getPath().getName())
+              .filter(n -> !fsFiles.containsKey(n)).collect(Collectors.toList());
+          if (!filesDeleted.isEmpty()) {
+            LOG.info("Found deleted files in partition " + partition + ": " + filesDeleted);
+            HoodieCleanPartitionMetadata cleanPartitionMetadata = new HoodieCleanPartitionMetadata(partition, "", filesDeleted, filesDeleted,
+                Collections.EMPTY_LIST, false);
+            partitionMetadata.put(partition, cleanPartitionMetadata);
+          }
+        }
+      }
+      cleanMetadata.setPartitionMetadata(partitionMetadata);
+
+      // Even if we don't have any deleted files to sync, we still create an empty commit so that we can track the restore has completed.
+      // We cannot create a deltaCommit at instantTime now because a future block (rollback) has already been written to the logFiles.
+      // We need to choose a timestamp which would be a validInstantTime for MDT. This is either a commit timestamp completed on the dataset
+      // or a timestamp with suffix which we use for MDT clean, compaction etc.
+      // String syncCommitTime = HoodieTableMetadataUtil.createIndexInitTimestamp(HoodieActiveTimeline.createNewInstantTime());
+      // TODO: Using METADATA_INDEXER_TIME_SUFFIX for now, but this should have its own suffix. To be fixed after HUDI-6200
+      String syncCommitTime = HoodieActiveTimeline.createNewInstantTime() + METADATA_INDEXER_TIME_SUFFIX;
```

Comment:
@codope : is there any dependencies we have around the indexing commit suffix? do you see any issues here. or can u confirm if we are good

Reply:
@codope : gentle ping